File: app/src/main/java/org/nevack/unitconverter/history/HistoryFilterDialog.java
package org.nevack.unitconverter.history;

import android.app.Dialog;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;
import androidx.appcompat.app.AlertDialog;

import org.nevack.unitconverter.model.EUnitCategory;

import java.util.ArrayList;
import java.util.List;

public class HistoryFilterDialog extends DialogFragment {



    public interface Listener {
        void OnResult(int result);
    }

    private Listener listener;
    private int mask = 0;

    public void setListener(Listener listener) {
        this.listener = listener;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        // Use the Builder class for convenient dialog construction
        AlertDialog.Builder builder = new AlertDialog.Builder(requireActivity());

        List<String> names = new ArrayList<>();
        names.add("None");
        for (EUnitCategory category : EUnitCategory.values()){
            names.add(getString(category.getName()));
        }


        builder.setItems(names.toArray(new String[0]), (dialog, which) -> mask = which - 1);
        builder.setOnDismissListener(dialog -> listener.OnResult(mask));

        // Create the AlertDialog object and return it
        return builder.create();
    }

}

File: app/src/main/java/org/nevack/unitconverter/history/HistoryFilterDialog.kt
package org.nevack.unitconverter.history

import android.os.Bundle
import android.app.Dialog
import java.util.ArrayList
import org.nevack.unitconverter.model.EUnitCategory
import android.content.DialogInterface
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment

typealias Listener = ((Int) -> Unit)

class HistoryFilterDialog : DialogFragment() {
    private var listener: Listener? = null
    private var mask = 0
    fun setListener(listener: Listener?) {
        this.listener = listener
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(requireContext())
        val names: MutableList<String> = mutableListOf("None")
        EUnitCategory.values().mapTo(names) { getString(it.getName()) }
        builder.setItems(names.toTypedArray()) { _: DialogInterface?, which: Int ->
            mask = which - 1
        }

        return builder.create()
    }

    override fun onDismiss(dialog: DialogInterface) {
        super.onDismiss(dialog)
        listener?.invoke(mask)
    }
}